rai/node/rpc.cpp

```diff
@@ -2869,6 +2869,64 @@ void rai::rpc_handler::search_pending_all ()
 	}
 }
 
+void rai::rpc_handler::request_reconfirmation ()
+{
+	std::string hash_text (request.get <std::string> ("hash"));
+	rai::block_hash hash_l;
+	if (!hash_l.decode_hex (hash_text))
+	{
+		rai::transaction transaction (node.store.environment, nullptr, false);
+		auto block_l (node.store.block_get (transaction, hash_l));
+		if (block_l != nullptr)
+		{
+			node.request_reconfirmation(std::shared_ptr<rai::block>(move(block_l)));
+		}
+		else
+		{
+			error_response (response, "Block not found");
+		}
+	}
+	else
+	{
+		error_response (response, "Invalid block hash");
+	}
+}
+
+void rai::rpc_handler::check_election_results ()
+{
+	std::string hash_text (request.get <std::string> ("hash"));
+	rai::block_hash hash_l;
+	if (!hash_l.decode_hex (hash_text))
+	{
+		rai::transaction transaction (node.store.environment, nullptr, false);
+		auto block_l (node.store.block_get (transaction, hash_l));
+		if (block_l != nullptr)
+		{
+			rai::election_result result;
+			if (!node.check_election_results (std::shared_ptr <rai::block>(move (block_l)), result))
+			{
+				boost::property_tree::ptree response_l;
+				response_l.put ("verified", result.met_reconfirm_threshold && result.confirmed);
+				response_l.put ("tally", result.tally.to_string());
+				response_l.put ("percent", (double)(result.tally.number() / (node.ledger.supply (transaction) / 1000)) / 10.0);
```

Comment:
Please qualify this as "percent_approved" or "percent_of_in_circulation_approved".

Reply:
Done.